[flake8] Fix flake8 errors

vchrombie opened the pull request:
This PR fixes the flake8 errors present in the muggle branch. This is necessary as a flake8 check must be added in the github action ci workflow.

It adds a configuration to ignore some errors of a file as it is hard to understand the logic if we format the file. It also adds a configuration to excludes a file.

vchrombie commented:
@sduenas, please review when you are free.
Thanks.

sduenas commented on sortinghat/cli/client/schema.py:
```diff
@@ -183,7 +183,8 @@ class CountryType(sgqlc.types.Type):
     code = sgqlc.types.Field(sgqlc.types.non_null(String), graphql_name='code')
```
We should exclude this file from the checks. It's going to keep failing because it is generated automatically.

vchrombie replied:
Okay. :+1:
The flake8 configuration file is missing anyway. I think we can add this file to the `exclude` list.

sduenas commented on sortinghat/core/schema.py:
```diff
@@ -826,10 +827,8 @@ def resolve_organizations(self, info, filters=None,
-                                 Q(name__in=Subquery(Domain.objects
-                                                    .filter(domain__icontains=search_term)
-                                                    .values_list('organization__name'))))
+            query = query.filter(Q(name__icontains=search_term) | Q(name__in=Subquery(
```
I rather keep this as in the original file. The query is easier to ready. Is there anyway to avoid this problem?

vchrombie replied:
Yes, we can do it. The flake8 documentation has a mention about it.
https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors

sduenas commented on sortinghat/core/schema.py:
```diff
@@ -845,20 +844,15 @@ def resolve_individuals(self, info, filters=None,
-                                                         Q(individual__profile__name__icontains=search_term) |
-                                                         Q(individual__profile__email__icontains=search_term))
-                                                 .values_list('individual__mk')))
+            query = query.filter(mk__in=Subquery(Identity.objects.filter(
```
Same as before.

sduenas reviewed (changes requested):
Please check my comments.

vchrombie commented:
Hi @sduenas, I updated the PR accordingly.
- added the file `sortinghat/cli/client/schema.py` file to exclude list
- added `# noqa: code` to the specific lines mentioned

sduenas commented:
> Hi @sduenas, I updated the PR accordingly.
> 
>     * added the file `sortinghat/cli/client/schema.py` file to exclude list
> 
>     * added `# noqa: code` to the specific lines mentioned

I really hate to have those comments around to avoid these problems. Maybe we should exclude this file too, to be honest.

vchrombie commented:
> I really hate to have those comments around to avoid these problems. Maybe we should exclude this file too, to be honest.

I had the same opinion but did it because the errors might pop in some other place. Well, I think we need not be that strict. I will add the file to the ignore list.

vchrombie commented:
I rebased the PR to include the latest changes. The build works too! :tada:

sduenas reviewed (approved):
LGTM
